lib/rules/blank-lines.js

```diff
@@ -19,30 +32,85 @@ module.exports = {
         docs: {
             recommended: true,
             type: "warning",
-            description: "Ensure that there is exactly a 2-line gap between Contract and Funtion declarations"
+            description: "Ensure that there is exactly a 2-line gap between Contract and Function declarations"
         },
 
+        fixable: "whitespace",
+
         schema: []
 
     },
 
     create: function(context) {
 
         let sourceCode = context.getSourceCode(),
-            noCodeRegExp = /^[ \t]*$/,
-            isCommentRegExp = /\s*(\/\/*)|(\*\\*)|(\/\**)$/;
+            noCodeRegExp = /^\s*$/,
+            isCommentRegExp = /\s*(\/\/*)|(\*\\*)|(\/\**)$/,
+            strictCommentRegExp = /^\s*(\/\/*)|(\*\\*)|(\/\**)$/; // Tests whether the entire line is a comment
         let topLevelDeclarations = [ "ContractStatement", "LibraryStatement" ];
 
         function inspectProgram(emitted) {
             if (emitted.exit) {
                 return;
             }
 
-            function report(topLevelNode) {
+            function report(topLevelNode, spacing) {
                 const message = `${topLevelNode.type.replace("Statement", "")} '${topLevelNode.name}' `
                     + "must be preceded by 2 blank lines.";
 
-                context.report({ node: topLevelNode, message });
+                context.report({
+                    node: topLevelNode, message,
+                    fix(fixer) {
+                        // The fix strategy is as follows:
+                        // - count lines going upwards (without going beyond the bounds of the spacing between this node and the previous one)
+                        // - Skip over any accompanying comments of the top level node
+                        // - Count the number of blank lines directly above the top-most accompanying comments iteratively
+                        // - Insert the desired number of line break characters right before the comment
+
+                        let line = sourceCode.getLine(topLevelNode) - 1, // The line number
+                            maxLines = spacing.split("\n").length,
+                            bottomLine = sourceCode.getLine(topLevelNode),
+                            multipleLines = spacing.indexOf("\n") > -1, // Whether there are multiple lines
```

Comment:
use `includes()` method in place of `indexOf(...) > -1` at all places where you need to check whether an item exists inside an array/string. The `indexOf()` is a bad habit I introduced long back when I wasn't aware of includes().
eg
```
// returns boolean
myArray.includes(90)
myString.includes("\n")
```